File: app/src/main/java/ru/wsr/stemobile/ui/main/SubstitutionLookup.java
package ru.wsr.stemobile.ui.main;

import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.selection.ItemDetailsLookup;
import androidx.recyclerview.widget.RecyclerView;


public class SubstitutionLookup extends ItemDetailsLookup<Long> {

    private final RecyclerView recyclerView;

    SubstitutionLookup(RecyclerView recyclerView) {
        this.recyclerView = recyclerView;
    }

    @Nullable
    @Override
    public SubstitutionItemDetail getItemDetails(@NonNull MotionEvent e) {
        View view = recyclerView.findChildViewUnder(e.getX(), e.getY());
        if (view != null) {
            RecyclerView.ViewHolder viewHolder = recyclerView.getChildViewHolder(view);
            if (viewHolder instanceof SubstitutionAdapter.SubstitutionViewHolder) {
                return ((SubstitutionAdapter.SubstitutionViewHolder) viewHolder).getItemDetails();
            }
        }
        return null;
    }
}

File: app/src/main/java/ru/mrlargha/stemobile/ui/main/SubstitutionLookup.kt
package ru.mrlargha.stemobile.ui.main

import android.view.MotionEvent
import androidx.recyclerview.selection.ItemDetailsLookup
import androidx.recyclerview.widget.RecyclerView
import ru.mrlargha.stemobile.ui.main.SubstitutionAdapter.SubstitutionViewHolder

class SubstitutionLookup internal constructor(private val recyclerView: RecyclerView
) : ItemDetailsLookup<Long?>() {
    override fun getItemDetails(e: MotionEvent): SubstitutionItemDetail? {
        val view = recyclerView.findChildViewUnder(e.x, e.y)
        if (view != null) {
            val viewHolder = recyclerView.getChildViewHolder(view)
            if (viewHolder is SubstitutionViewHolder) {
                return viewHolder.itemDetails
            }
        }
        return null
    }

}